table_schema_cache.go

```diff
@@ -342,6 +342,75 @@ func (c TableSchemaCache) GetTableListWithPriority(priorityList []string) (prior
 	return
 }
 
+func (c TableSchemaCache) GetTableCreationOrder(db *sql.DB) ([]string, error) {
+	visitedTables := map[string]bool{}
+	tableMap := map[string][]string{}
+	tableOrder := []string{}
+
+	for table := range c {
+		t := strings.Split(table, ".")
+		referencedTables, err := getForeignKeyTablesOfTable(db, t[0], t[1])
+		if err != nil {
+			logrus.WithField("table", table).Error("cannot fetch foreign keys")
+		}
+		tableMap[table] = referencedTables
+	}
+
+	for table := range c {
+		if _, found := visitedTables[table]; !found {
+			visitedTables, tableOrder = getTableCreationOrderUtil(visitedTables, tableOrder, tableMap, table)
```

Comment:
Can you document how this algorithm is supposed to work (in either comments attached to a function here, or via a separate document)? It looks feasible to trace through myself, but it's better for a non-trivial algorithm to have its intent documented, so reviewers can focus on if the implementation is right or wrong, not reverse engineering what you're trying to do.